### ISSUE TITLE
LLDB type summaries for units classes

### ISSUE THREAD

**Issue by gyrovague** [nholthaus/units#279]
Has anybody tried making type summaries for units classes, so that the values of units variables can be easily displayed during debugging?

e.g. see:
https://lldb.llvm.org/use/variable.html#type-summary
https://pspdfkit.com/blog/2018/how-to-extend-lldb-to-provide-a-better-debugging-experience/

**Comment by nholthaus** [comment]
No. I tried a .natvis file once but never really took it anywhere. I think
summaries would be a useful addition and I'd certainly take a PR for it.

**Comment by nholthaus** [comment]
Shipped in [v3.5.0](https://github.com/nholthaus/units/releases/tag/v3.5.0). `natvis/units_lldb.py` is an LLDB summary provider that renders a quantity as its value and abbreviation (`5 m`, `3 s`) in LLDB and its front-ends (CLion, Xcode, VS Code CodeLLDB), mirroring the existing MSVC natvis. Load it with `command script import .../natvis/units_lldb.py` (or from your `~/.lldbinit`); it's shipped in the installed package. It's general — one regex-matched summary covers every unit type, and named units render through their `unit<...>` base. Details in the [debugger-visualizers how-to](https://github.com/nholthaus/units/blob/main/docs/how-to/natvis.md).
